test/rdoc/test_rdoc_markup_to_html_crossref.rb

```diff
@@ -30,6 +33,21 @@ def test_convert_CROSSREF
     assert_equal para("<code># :stopdoc:</code>:"), result
   end
 
+  def test_convert_CROSSREF_ignored_excluded_words
+    @options.autolink_excluded_words = ['C1']
+
+    result = @to.convert 'C1'
+
+    assert_equal para("C1"), result
+
+    result = @to.convert '+C1+'
+    assert_equal para("<code>C1</code>"), result
```

Comment:
Wouldn't it be better that this is linked?
Simple word like "Ruby", "Set" and "Rails" may not be classes/modules, but code like `+Ruby+`, `+Set+` and `+Rails+` (or `` `Ruby` ``, `` `Set` `` and `` `Rails` `` in Markdown format) are very likely classes/modules, I think.

Reply:
While I understand the argument, I think it makes the behaviour harder to understand for users. If both `C1` and `+C1+` are autolinked, I'd expect the exclusion to happen to both of them as well.

Also, some people, including me, often use markdown's code to highlight & emphasize certain keywords without expecting linking. And that causes examples like [this](https://docs.ruby-lang.org/en/master/contributing/building_ruby_md.html):

<img width="70%" alt="Screenshot 2024-12-29 at 11 48 19" src="https://github.com/user-attachments/assets/377b3b5f-1529-4e0b-93ec-168267d19a11" />

So I'd expect this config to fix that behavior too.

Reply:
> While I understand the argument, I think it makes the behaviour harder to understand for users. If both `C1` and `+C1+` are autolinked, I'd expect the exclusion to happen to both of them as well.

These are different.
`C1` does not need to be the whole code,  it is auto-linked in a sentence `the C1 class`.
However `+C1+` must be the whole code, `+the C1 class+` is not linked.

I mean to limit the config to plain texts.

> Also, some people, including me, often use markdown's code to highlight & emphasize certain keywords without expecting linking. And that causes examples like [this](https://docs.ruby-lang.org/en/master/contributing/building_ruby_md.html):
> <img alt="Screenshot 2024-12-29 at 11 48 19" width="70%" src="https://private-user-images.githubusercontent.com/5079556/399157541-377b3b5f-1529-4e0b-93ec-168267d19a11.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzU0NzM3MjAsIm5iZiI6MTczNTQ3MzQyMCwicGF0aCI6Ii81MDc5NTU2LzM5OTE1NzU0MS0zNzdiM2I1Zi0xNTI5LTRlMGItOTNlYy0xNjgyNjdkMTlhMTEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIyOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMjlUMTE1NzAwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OTdiOGE0ODA3MzY0OTQ2YzQ5NGExZTAzY2E4NDdjZDE1Yzg5ZjA1Nzk3N2ZlYTRmMWY4ZDdkNTM1ZmVkMWE1ZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.HLI9cQJH193r8jyPC-pyv_jb6OggbLkLeW7NTC6idCw">

The word "Ruby" is not marked as code in that example, is it?
These "Ruby" will not be linked in my proposal.

(and I think we should discourage use of markdown's code just to highlight & emphasize non-code)

> So I'd expect this config to fix that behavior too.

I'm not against the config itself.
But a new directive at that module feels better to me.

Reply:
```diff
diff --git i/lib/rdoc/markup/to_html_crossref.rb w/lib/rdoc/markup/to_html_crossref.rb
index 76714522..678e5ee1 100644
--- i/lib/rdoc/markup/to_html_crossref.rb
+++ w/lib/rdoc/markup/to_html_crossref.rb
@@ -92,6 +92,8 @@ class RDoc::Markup::ToHtmlCrossref < RDoc::Markup::ToHtml
       return name if name =~ /\A[a-z]*\z/
     end
 
+    return name if @options.autolink_excluded_words&.include?(name)
+
     cross_reference name, rdoc_ref: false
   end
 
@@ -151,8 +153,6 @@ class RDoc::Markup::ToHtmlCrossref < RDoc::Markup::ToHtml
       label = $'
     end
 
-    return name if name == text && @options.autolink_excluded_words&.include?(name)
-
     ref = @cross_reference.resolve name, text if name
 
     case ref
diff --git i/test/rdoc/test_rdoc_markup_to_html_crossref.rb w/test/rdoc/test_rdoc_markup_to_html_crossref.rb
index cf95ebf3..e50002e8 100644
--- i/test/rdoc/test_rdoc_markup_to_html_crossref.rb
+++ w/test/rdoc/test_rdoc_markup_to_html_crossref.rb
@@ -41,7 +41,7 @@ class RDocMarkupToHtmlCrossrefTest < XrefTestCase
     assert_equal para("C1"), result
 
     result = @to.convert '+C1+'
-    assert_equal para("<code>C1</code>"), result
+    assert_equal para("<a href=\"C1.html\"><code>C1</code></a>"), result
 
     # Explicit linking with rdoc-ref is not ignored
     result = @to.convert 'Constant[rdoc-ref:C1]'
```

Reply:
So in the case of `+C1+`, it's an explicit cross-ref linking syntax now. Can you update this [document](https://github.com/ruby/rdoc/blob/master/doc/rdoc/markup_reference.rb) to mention it?

Reply:
I've updated the implementation to follow this expectation.